include/knowhere/comp/prometheus_client.h

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#pragma once
+
+#include <prometheus/collectable.h>
+#include <prometheus/counter.h>
+#include <prometheus/exposer.h>
+#include <prometheus/gauge.h>
+#include <prometheus/histogram.h>
+#include <prometheus/registry.h>
+#include <prometheus/summary.h>
+#include <prometheus/text_serializer.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "knowhere/log.h"
+
+namespace knowhere {
+
+using Quantiles = std::vector<prometheus::detail::CKMSQuantiles::Quantile>;
+using BucketBoundaries = std::vector<double>;
+
+#define PROMETHEUS_GAUGE(name_gauge, name_family, name, desc, registry)    \
+    prometheus::Family<prometheus::Gauge>& name_family =                   \
+        prometheus::BuildGauge().Name(name).Help(desc).Register(registry); \
+    prometheus::Gauge& name_gauge = name_family.Add({});
+
+#define PROMETHEUS_COUNTER(name_counter, name_family, name, desc, registry)  \
+    prometheus::Family<prometheus::Counter>& name_family =                   \
+        prometheus::BuildCounter().Name(name).Help(desc).Register(registry); \
+    prometheus::Counter& name_counter = name_family.Add({});
+
+#define PROMETHEUS_SUMMARY(name_summary, name_family, name, desc, quantiles, registry) \
+    prometheus::Family<prometheus::Summary>& name_family =                             \
+        prometheus::BuildSummary().Name(name).Help(desc).Register(registry);           \
+    prometheus::Summary& name_summary = name_family.Add({}, quantiles);
+
+#define PROMETHEUS_HISTOGRAM(name_histogram, name_family, name, desc, boundaries, registry) \
+    prometheus::Family<prometheus::Histogram>& name_family =                                \
+        prometheus::BuildHistogram().Name(name).Help(desc).Register(registry);              \
+    prometheus::Histogram& name_histogram = name_family.Add({}, boundaries);
+
+template <typename T>
+using Family = ::prometheus::Family<T>;
+
+using Gauge = ::prometheus::Gauge;
+using Counter = ::prometheus::Counter;
+using Histogram = ::prometheus::Histogram;
+using Summary = ::prometheus::Summary;
```

Comment:
Why we set these aliases here? It is a bad practice to use alias in a .h file

Reply:
removed